pg_rational.c

```diff
@@ -111,32 +111,47 @@ rational_in_float(PG_FUNCTION_ARGS)
 {
 	float8		target = PG_GETARG_FLOAT8(0),
 				z,
-				prev_denom,
+				fnumer,
+				fdenom,
 				error;
 	int32		temp,
+				prev_denom,
 				sign;
 	Rational   *result = palloc(sizeof(Rational));
 
-	if (target == floor(target))
+	if (target == (int32)target)
 	{
-		result->numer = floor(target);
+		result->numer = (int32)target;
```

Comment:
Wondering if the explicit cast is required. Since numer was declared as int32 will the assignment automatically do the conversion anyway?